src/sticky.js

```diff
@@ -147,59 +173,73 @@ export default class Sticky extends React.Component {
-      if (newState.distanceFromBottom !== state.distanceFromBottom) return true;
+      if (newState.placeholderTop !== state.placeholderTop) return true;
     }
 
+    // We should check container sizes anyway
+    if (newState.containerOffset !== state.containerOffset) return true;
+    if (newState.containerBottom !== state.containerBottom) return true;
+    if (newState.containerTop !== state.containerTop) return true;
+
     return false;
   }
 
+  getPositionOffset() {
+    const { containerOffset, containerTop, containerBottom, height } = this.state;
+    const { bottomOffset, position, topOffset } = this.props;
+
+    const bottomLimit = containerBottom - height - bottomOffset
+    const topLimit =  window.innerHeight - containerTop - topOffset
+
+    return position === 'top'
+      ? Math.min(containerOffset, bottomLimit)
+      : Math.min(containerOffset, topLimit)
+  }
+
   /*
    * The special sauce.
    */
   render() {
-    const placeholderStyle = { paddingBottom: 0 };
-    let className = this.props.className;
+    const {
+      className: propsClassName,
+      position,
+      stickyClassName,
+      stickyStyle,
+      style,
+      ...props
+    } = this.props;
+    const {
+      isSticky,
+      height,
+      width,
+      xOffset,
+    } = this.state
+
+    const placeholderStyle = { paddingBottom: isSticky ? height : 0 };
+    const className = `${propsClassName} ${isSticky ? stickyClassName : ''}`;
+    const finalStickyStyle = isSticky && {
+      position: 'fixed',
+      [position]: this.getPositionOffset(),
+      left: xOffset,
+      width: width,
+      ...stickyStyle,
+    }
 
     // To ensure that this component becomes sticky immediately on mobile devices instead
     // of disappearing until the scroll event completes, we add `transform: translateZ(0)`
     // to 'kick' rendering of this element to the GPU
     // @see http://stackoverflow.com/questions/32875046
-    let style = Object.assign({}, { transform: 'translateZ(0)' }, this.props.style);
-
-    if (this.state.isSticky) {
-      const stickyStyle = {
-        position: 'fixed',
-        top: this.state.containerOffset,
-        left: this.state.xOffset,
-        width: this.state.width
-      };
-
-      const bottomLimit = this.state.distanceFromBottom - this.state.height - this.props.bottomOffset;
-      if (this.state.containerOffset > bottomLimit) {
-        stickyStyle.top = bottomLimit;
-      }
-
-      placeholderStyle.paddingBottom = this.state.height;
-
-      className += ` ${this.props.stickyClassName}`;
-      style = Object.assign({}, style, stickyStyle, this.props.stickyStyle);
+    const finalStyle = {
+      transform: 'translateZ(0)',
+      ...style,
+      ...(finalStickyStyle || {})
     }
 
-    const {
-      topOffset,
-      isActive,
-      stickyClassName,
-      stickyStyle,
-      bottomOffset,
-      onStickyStateChange,
-      ...props
-    } = this.props;
```

Comment:
Removing this has caused a regression as extra props are being passed to the `div` again - #99